### ISSUE TITLE
Missing type parameters in MethodDeclaration.getDeclarationAsString()

### ISSUE THREAD

**Issue by Symbolk** [javaparser/javaparser#2131]
Sometimes we need to get exactly the original signature of method declaraion, but I found calling 
getDeclarationAsString() misses type parameters (e.g. <R, A>).

Just like modifiers/throws/parameter names, whether to include type parameters should be optional too, considering that it is between modifiers and the return type.

https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/ast/body/MethodDeclaration.java

**Comment by matozoid** [comment]
I'll mark this as "bug" until I get time to figure out if they are supposed to be there or not.

**Comment by matozoid** [comment]
I don't have a clue. @ftomassetti?

**Comment by ftomassetti** [comment]
I do not remember how this one was intended to work

**Comment by Symbolk** [comment]
> I do not remember how this one was intended to work

Yeah

**Comment by Symbolk** [comment]
Yeah, I just mean the type parameters should be optional, just like the other elements composing method declaration.

**Comment by matozoid** [comment]
That seems like a decent feature request.

**Comment by matozoid** [comment]
I guess we can refactor these methods to use a visitor too, instead of patching a string together.
